datafusion/core/tests/sqllogictests/test_files/select.slt

```diff
@@ -230,3 +230,10 @@ select CASE
 END;
 ----
 2
+
+# like and ilike for LargeUtf8
+# issue: https://github.com/apache/arrow-datafusion/issues/5893
+query B
+select arrow_cast('foo', 'LargeUtf8') like '%foo%';
```

Comment:
cool, btw, why is it like '%foo%', not == 'foo' ?

Reply:
It's a case of issue

Reply:
> It's a case of issue

Sorry, I still didn't get it, you mean inner value of Utf8("foo") not the same as inner value LargeUtf8("foo")?

Reply:
Sorry, it's not clear.
This test cover issue in #5893

Reply:
I guess a more comprehensive test would be 
```
select arrow_cast('foobar', 'LargeUtf8') like '%oba%';
—-
true


select arrow_cast('FooBar', 'LargeUtf8') like '%oba%';
—-
true
```


https://www.postgresql.org/docs/7.3/functions-matching.html#:~:text=The%20keyword%20ILIKE%20can%20be,and%20~~*%20corresponds%20to%20ILIKE%20.

Reply:
Another test that would be great to add would be to compare columns as well as constants (as that is the other binary expr codepath)

Something like

```sql
create table t as values 
  (arrow_cast('Foo', 'LargeUtf8') '%f'), 
  (arrow_cast('Bar', 'LargeUtf8') 'B%');

select column1 like column2 from t;
select column1 ilike column2 from t;
select column1 not like column2 from t;
select column1 not ilike column2 from t;
```